cores/esp32/esp32-hal-i2c.c

```diff
@@ -1628,6 +1631,9 @@ i2c_err_t i2cWrite(i2c_t * i2c, uint16_t address, uint8_t* buff, uint16_t size,
 }
 
 i2c_err_t i2cRead(i2c_t * i2c, uint16_t address, uint8_t* buff, uint16_t size, bool sendStop, uint16_t timeOutMillis, uint32_t *readCount){
```

Comment:
if size is 0, why return an error? if size is not zero, but buf is null, then I understand, but size 0 is fine to return OK

Reply:
A zero byte read creates an invalid I2C bus state.  Depending on what the first bit the SLAVE responds with, the i2c peripheral may not be able to complete the transmission.  A zero byte write is possible, but a zero byte read is not allowed.  The zero byte write is used as a presence detect event.:
```c++
Wire.beginTransmission(ID);
uin8_t error = Wire.endTransmission();
if(error == 2) { // slave did not answer, is not present
  Serial.printf("NAK, no slave present at 0x%02X\n",ID);
} else if (error == 0) {// Slave ready for communications
  Serial.printf("ACK, slave present at 0x%02X\n",ID);
}
```
The following code is invalid:
```c++
uint8_t requestSize = 0;
uint8_t count = Wire.requestFrom(ID,requestSize);
if( count == requestSize) {// Slave is present and ready for communications
   Serial.printf("ACK, slave present at 0x%02X, received %dbytes\n",ID,count);
   Serial.printf(" requestFrom succeeded =%d(%s)\n",Wire.lastError(),Wire.getErrorText(Wire.lastError()));
   for (uint8_t i = 0 ; i< count; i++){
     Serial.print(Wire.read());
  }
  Serial.println():
}
else { // problem
   Serial.printf(" requestFrom Failed =%d(%s)\n",Wire.lastError(),Wire.getErrorText(Wire.lastError()));
}
```
change `requestSize` to a non zero value and the code works a expected, 
The act of requesting a zero byte read must error out. 

A Zero byte Write will return information, Whether a Slave device is active on the bus.

A Zero byte Read is meaningless. and will only succeed if the correct circumstances exist, else it will hang the i2c bus creating a error cascade that is confusing.

Chuck.

Reply:
ok...